salt/modules/crmsh.py

```diff
@@ -9,3 +9,108 @@
 :depends: crmsh
 '''
 from __future__ import absolute_import
+import re
+
+from salt.ext import six
+from salt.exceptions import CommandExecutionError
+
+
+def _dc():
+    '''
+    check whether we are DC node
+    '''
+    import socket
+
+    try:
+        res = __salt__['cmd.run_all']("crmadmin -D")
+    except:
+        return False
+    if 'stdout' in res.keys():
+        dc = res['stdout'].split()[-1]
+        if dc != socket.gethostname():
+            return False
+
+    return True
+
+
+def configure_property(prop):
+    '''
+    Set a cluster property
+
+    prop
+        format of each property should be \"key=value\", or a dict type
+
+    This function will reject to run if current node is not DC.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' crmsh.configure_property prop='stonith-enabled=true'
+    '''
+    if not _dc():
+        raise CommandExecutionError("This function can only run at DC node")
+
+    cmd = ['crm', 'configure', 'property']
+
+    if isinstance(prop, dict):
```

Comment:
When is it invoked with a dict as argument, does salt do that somehow?

Reply:
I have no idea other modules  also use dict as argument,
I just thought for user convenience, both dict and string type argument are supported